pkg/events/events.go

```diff
@@ -164,6 +195,31 @@ func New(ctx context.Context, name string, identifiers CombinedIdentifiers, data
 	if identifiers != nil {
 		evt.innerEvent.Identifiers = identifiers.CombinedIdentifiers().GetEntityIdentifiers()
 	}
+	authentication := &ttnpb.Event_Authentication{}
+	if p, ok := peer.FromContext(ctx); ok && p.Addr.String() != "pipe" {
+		host, _, err := net.SplitHostPort(p.Addr.String())
+		if err == nil {
+			authentication.RemoteIP = host
+		}
+	}
+	md := rpcmetadata.FromIncomingContext(ctx)
+	if md.AuthType != "" {
+		authentication.Type = md.AuthType
+	}
+	if md.AuthValue != "" {
+		// Cannot use auth.SplitToken() here because of import cycle
+		parts := strings.Split(md.AuthValue, ".")
+		if len(parts) == 3 {
+			authentication.TokenType = parts[0]
+			authentication.TokenID = parts[1]
+		}
+	}
+	if md.XForwardedFor != "" {
+		authentication.RemoteIP = md.XForwardedFor
+	}
```

Comment:
The value of the XFF header isn't equal to RemoteIP. We may have to split it and take the correct one if there are multiple.

Reply:
How is "the correct one" defined in this context?

Reply:
Updated to only keep the first part of the X-forwarded-for header (client IP)